common/types/shared.go

```diff
@@ -3646,64 +3646,16 @@ func (v *History) GetEvents() (o []*HistoryEvent) {
 
 // HistoryBranch is an internal type (TBD...)
 type HistoryBranch struct {
-	TreeID    *string               `json:"treeID,omitempty"`
-	BranchID  *string               `json:"branchID,omitempty"`
-	Ancestors []*HistoryBranchRange `json:"ancestors,omitempty"`
-}
-
-// GetTreeID is an internal getter (TBD...)
-func (v *HistoryBranch) GetTreeID() (o string) {
-	if v != nil && v.TreeID != nil {
-		return *v.TreeID
-	}
-	return
-}
-
-// GetBranchID is an internal getter (TBD...)
-func (v *HistoryBranch) GetBranchID() (o string) {
-	if v != nil && v.BranchID != nil {
-		return *v.BranchID
-	}
-	return
-}
-
-// GetAncestors is an internal getter (TBD...)
-func (v *HistoryBranch) GetAncestors() (o []*HistoryBranchRange) {
-	if v != nil && v.Ancestors != nil {
-		return v.Ancestors
-	}
-	return
+	TreeID    string
+	BranchID  string
+	Ancestors []*HistoryBranchRange
```

Comment:
I don't really think it should delay this PR, but: can we drop the pointers here?  I doubt a nil item in the slice is desirable.
(unless it is.  then LGTM)

Reply:
In general I agree, but this would end up touching a lot more code, therefore I will keep it for now.